etcdserver/server.go

```diff
@@ -204,6 +204,13 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
 		cl *cluster
 	)
 
+	if mkerr := fileutil.MkdirAllIfNotExist(cfg.DataDir); mkerr != nil {
+		return nil, fmt.Errorf("cannot create data directory: %v", mkerr)
+	}
+	if wderr := fileutil.IsDirWriteable(cfg.DataDir); wderr != nil {
```

Comment:
I had a hard time to think of a good name for it.

Basically there are 4 cases:

1. the dir does not exist; a dir is made; the dir must be writable

2. the dir does not exist; a dir fails to be made; the dir must be not writable

3. the dir exists, check the dir and it is writable

4. the dir exists, check the dir and it is not writable

I cannot come up with a good func name to summary these functionality.